File: docs/examples/py/src/access_control_browse_tokens.py
from typing import List

from reduct import Client, Permissions, Token, FullTokenInfo


async def browse_tokens():
    # Create a client with the base URL and API token
    async with Client("http://localhost:8383", api_token="my-token") as client:
        #  Browse all tokens and print their information
        tokens: List[Token] = await client.get_token_list()
        for token in tokens:
            print(f"Token: {token.name}")
            print(f"Created: {token.created_at}")
            print(f"Provisioned: {token.is_provisioned}")

            # Get detailed information about a specific token
            details: FullTokenInfo = await client.get_token(token.name)
            print(f"Permissions: {details.permissions}")


if __name__ == "__main__":
    import asyncio

    asyncio.run(browse_tokens())

File: versioned_docs/version-1.21.x/examples/js/src/access_control_browse_tokens.mjs
import { Client } from "reduct-js";

// Create a new client with the server URL and an API token
const client = new Client("http://127.0.0.1:8383", { apiToken: "my-token" });

// Browse all tokens and print their information
const tokenList = await client.getTokenList();
for (const token of tokenList) {
  console.log(`Token: ${token.name}`);
  console.log(`Created: ${token.createdAt}`);
  console.log(`Provisioned: ${token.isProvisioned}`);

  // Get detailed information about a specific token
  const tokenInfo = await client.getToken(token.name);
  console.log(`Permissions: ${tokenInfo.permissions}`);
}
